src/NGen.cpp

```diff
@@ -754,8 +750,71 @@ int main(int argc, char* argv[]) {
 #endif
 
 #if NGEN_WITH_ROUTING
+#if NGEN_WITH_MPI
+    if (mpi_num_procs > 1) {
+        int number_of_timesteps = manager->Simulation_Time_Object->get_total_output_times();
+        std::vector<std::string> local_nexus_ids;
+        for (const auto& nexus : nexus_indexes) {
+            local_nexus_ids.push_back(nexus.first);
+        }
+        // MPI_Gather all nexus IDs into a single vector
+        std::vector<std::string> all_nexus_ids = parallel::gather_strings(local_nexus_ids, mpi_rank, mpi_num_procs);
+        if (mpi_rank == 0) {
+            // filter to only the unique IDs
+            std::sort(all_nexus_ids.begin(), all_nexus_ids.end());
+            all_nexus_ids.erase(
+                std::unique(all_nexus_ids.begin(), all_nexus_ids.end()),
+                all_nexus_ids.end()
+            );
+        }
+        // MPI_Broadcast so all processes share the nexus IDs
+        all_nexus_ids = std::move(parallel::broadcast_strings(all_nexus_ids, mpi_rank, mpi_num_procs));
+
+        // MPI_Reduce to collect the results from processes
+        std::vector<double> all_nexus_downflows;
+        if (mpi_rank == 0) {
+            all_nexus_downflows.resize(number_of_timesteps * all_nexus_ids.size(), 0.0);
+        }
+        std::unordered_map<std::string, int> all_nexus_indexes;
+        std::vector<double> local_buffer(number_of_timesteps);
+        std::vector<double> receive_buffer(number_of_timesteps, 0.0);
+        for (int i = 0; i < all_nexus_ids.size(); ++i) {
+            std::string nexus_id = all_nexus_ids[i];
+            if (nexus_indexes.find(nexus_id) != nexus_indexes.end()) {
```

Comment:
Given that we're going to this effort to gather and reduce the values of nexus flows across processes, I realize that much of the existing MPI logic in ngen actually becomes redundant as a result, and I think the code here (maybe with some eventual refactoring) is preferable.

For the simplest modification to just have this all work correctly for now, I think this should do it:

```suggestion
            if (nexus_indexes.find(nexus_id) != nexus_indexes.end() && !features.is_remote_sender_nexus(id)) {
```

Basically, beyond just "this process interacts with this nexus", it's also "this process receives/records all of the flows into this nexus", as in `SurfaceLayer.cpp`.

In the long run, we could do all the accumulation and output (as necessary) from here, and delete `HY_PointHydroNexusRemote` entirely.